### PR TITLE
Ft components

### PR DESCRIPTION
**Pull request by memelopez**
## This PR implements: 👨‍💻 🆕 🔥 

- no [linter errors](https://github.com/microverseinc/linters-config) ✅ 
- correct use of [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md) ✅ 
- work is documented in a [professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md) ✅ 
- follows [JS practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md) ☑️ 
- **deleted all the boilerplate** from CRA ✔️ 
- **created new directory** called `components` ✔️ 
- inside `components`, created a new `Calculator.js` file ✔️ 
- in `Calculator.js`, cerated a react component that matches the design ✔️ 
- implemented **desgin only** ✔️ 
- used **functional** components ✔️ 
- created **1 child component** ✔️ 
- once the `Calculator` was ready, imported it to **be used in top-level component**: `App.js` ✔️ 
- **Tested in browser**, runs without errors ✔️

### CODE REVIEW

**Review by AmaduKamara — COMMENTED** [review]
# Status: Invalid Submission ❌ ❌ ❌

Hi @memelopez,

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that does not have the recommended README set up.

- [x] Kindly use this [README Template](https://github.com/microverseinc/readme-template) for your project and edit its content to match your project and then submit for review 👍🏽

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._
```

File: src/components/Calculator.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+Output.propTypes = {
+  result: PropTypes.number,
+};
+
+class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+    this.calculation = 0;
+  }
```
Comment (Reem-lab) [inline src/components/Calculator.js]: - [x] Great job working so far but I think you missed to make your calculator component as functional component as I can see here it is class based so please make it functional based as required 😊✨🔥🤩
image

**Review by Reem-lab — CHANGES_REQUESTED** [review]
# Hi @memelopez, 👋🏻👋🏻 

Great job so far! 💯🔥♻
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- No Linters erros 💯
- Followed Gitflow ✅
- Great working on the fuctionality ✅
- Descriptive PR 💯


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: .eslintrc.json [hunk 2]
```diff
@@ -18,7 +18,9 @@
     "react/jsx-filename-extension": ["warn", { "extensions": [".js", ".jsx"] }],
     "react/react-in-jsx-scope": "off",
     "import/no-unresolved": "off",
-    "no-shadow": "off"
+    "no-shadow": "off",
+    "no-useless-constructor": 0,
+    "@typescript-eslint/no-useless-constructor": 0
```
Comment (emmiiorji) [inline .eslintrc.json]: Please remove the added lines to the `.eslintrc.json` file.

**Review by emmiiorji — CHANGES_REQUESTED** [review]
Hi @memelopez, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

<div align='center'>

</div>

You have submitted a project with modified linters configuration. Kindly avoid modifying linter configurations because they represent coding standards that you share with your team.

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by frank1738 — COMMENTED** [review]
Hi @memelopez , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

- [x] Your readme file is missing some important headers like the `key features` and `future features`. Kindly add them to  make your readme file more descriptive

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by memelopez** [comment]
with all due respect, I took out the key features. It's just a calculator that won't even calculate right now... Is it really that important? please, it's the fourth time I've tried to submit this simple project, and 4 times I've been marked as invalid, 1 was understandable (reem-lab's) the other 3 times are nonsense. TWO TIMES HAVE BEEN ABOUT THE README FILEE!!! please.... Im getting kind of annoyed here 😮‍💨

File: src/components/Calculator.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+//           <Button className="calcBtn" value="3" />
+//           <Button className="calcBtn" value="+" />
+//           <Button className="calcBtn" value="0" />
+//           <Button className="calcBtn" value="." />
+//           <Button value="=" />
+//         </div>
+//       </div>
+//     );
+//   }
+// }
```
Comment (NduatiKagiri) [inline src/components/Calculator.js]: - [x] To make sure that your code is clean, please remove this unused code.

**Review by NduatiKagiri — CHANGES_REQUESTED** [review]
Hi @memelopez 👋 ,

# CHANGES REQUIRED 💯  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost-there

## Highlights ✔️ 
✅ No Linters errors
✅ Gitflow observed

## Required Changes ♻️

- [x] Your project has met the requirements and is almost complete. However, for your project to be approved, kindly make the change requested by the previous reviewer and complete adding the `key features` and `future features` headers in your README file to make it more descriptive.

- [x] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by memelopez** [comment]
@frank1738 @NduatiKagiri [here](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md)  is the project requirements for documenting the project in a professional way. It says that I can customize the template to my project. That my project was marked invalid even before to checking the code, shows how you are looking for any pretext to block my progress.

**Comment by joskalenda** [comment]
> @frank1738 @NduatiKagiri [here](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md) is the project requirements for documenting the project in a professional way. It says that I can customize the template to my project. That my project was marked invalid even before to checking the code, shows how you are looking for any pretext to block my progress.

Hey @memelopez, I have read your comments and I wish to say something about them, first be aware that we do not gain any advantage in blocking your progress, but what we want is to help you understand a good way of working and communicating in a big team. you are the student and we are reviewers, we check again the code you have checked already, and we can't wait to request changes if there is something that is meant to change. it is for your own good.

- [ ] The second point I want to point out is the way you are addressing your opinion, you are in math magician, the 1st weeks at microverse, we have been taught how to interact with others in morning sessions, standup teams, and pair groups, I can feel that `(.  ...)` in your word and it is not good at all, consider this program as work environment, you do not know everything, allow others to help and guide you too, either you know or not, `humility is an asset`. all reviewers were right, you should ask questions instead of complaining dear @memelopez

**Review by joskalenda — APPROVED** [review]
| STATUS: APPROVED ✅ |
|------------------------|
| Hi @memelopez , 🥰|
| Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉  |
|  |

### Highlights
✅ ... Green linters

---------
### Optional suggestions

> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### Cheers and Happy coding!👏👏👏
-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**